lisa/tools/kdump.py

```diff
@@ -0,0 +1,206 @@
+                "Kdump is not supported."
+            )
+
+    def config_crashkernel_memory(
+        self,
+        crashkernel: str,
+    ) -> None:
+        # For Redhat and Suse, config crashkernel in /etc/default/grub
+        # For Ubuntu/Debian, the /etc/default/grub.d/kdump-tools.cfg will override the
+        # configuration in /etc/default/grub, so we config crashkernel in
+        # kdump-tools.cfg.
+        cfg_file = ""
+        if isinstance(self.node.os, Redhat):
+            if self.node.os.information.version < "8.0.0":
+                cfg_file = "/etc/default/grub"
+                if self.node.shell.exists(PurePosixPath("/sys/firmware/efi")):
+                    cmd = "grub2-mkconfig -o /boot/efi/EFI/redhat/grub.cfg"
+                else:
+                    cmd = "grub2-mkconfig -o /boot/grub2/grub.cfg"
+            else:
+                cmd = f'grubby --update-kernel=ALL --args="crashkernel={crashkernel}"'
+        elif isinstance(self.node.os, Suse):
+            cfg_file = "/etc/default/grub"
+            cmd = "grub2-mkconfig -o /boot/grub2/grub.cfg"
+        elif isinstance(self.node.os, Debian):
+            kdump_cfg = "/etc/default/grub.d/kdump-tools.cfg"
+            if self.node.shell.exists(PurePosixPath(kdump_cfg)):
+                cfg_file = kdump_cfg
+            else:
+                cfg_file = "/etc/default/grub"
+            cmd = "update-grub"
+
+        if cfg_file:
+            sed = self.node.tools[Sed]
+            result = self.node.execute(
+                f"grep crashkernel {cfg_file}", shell=True, sudo=True
+            )
+            if result.exit_code != 0:
+                sed.substitute(
+                    "^GRUB_CMDLINE_LINUX_DEFAULT",
+                    '"$',
+                    f"crashkernel={crashkernel}",
+                    cfg_file,
+                    sudo=True,
+                )
+            else:
+                sed.substitute(
+                    "^GRUB_CMDLINE_LINUX_DEFAULT",
+                    'crashkernel=[^[:space:]"]*',
+                    f"crashkernel={crashkernel}",
+                    cfg_file,
+                    sudo=True,
+                )
+
+        # Update grub
+        result = self.node.execute(cmd, shell=True, sudo=True)
+        result.assert_exit_code(message="Failed to update grub")
+
+    def enable_kdump_service(self) -> None:
+        systemctl = self.node.tools[Systemctl]
+        if isinstance(self.node.os, Redhat) or isinstance(self.node.os, Suse):
+            systemctl.enable_service("kdump")
+        elif isinstance(self.node.os, Debian):
+            systemctl.enable_service("kdump-tools")
+
+    def set_unknown_nmi_panic(self) -> None:
+        nmi_panic_file = "/proc/sys/kernel/unknown_nmi_panic"
+        if self.node.shell.exists(PurePosixPath(nmi_panic_file)):
+            result = self.node.execute(
+                "sysctl -w kernel.unknown_nmi_panic=1", sudo=True
```

Comment:
add a tool for sysctl, it should be used frequently.

Reply:
Have added.